src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/CosmosTemplate.java

```diff
@@ -294,48 +252,46 @@ public String getCollectionName(Class<?> domainClass) {
     public CosmosContainerProperties createCollectionIfNotExists(@NonNull CosmosEntityInformation<?, ?> information) {
         final CosmosContainerResponse response = cosmosClient
                 .createDatabaseIfNotExists(this.databaseName)
+                .onErrorResume(throwable ->
+                    exceptionHandler("Failed to create database", throwable))
                 .flatMap(cosmosDatabaseResponse -> {
                     fillAndProcessResponseDiagnostics(responseDiagnosticsProcessor,
                         cosmosDatabaseResponse, null);
                     return cosmosDatabaseResponse
                         .database()
                         .createContainerIfNotExists(information.getCollectionName(),
                             "/" + information.getPartitionKeyFieldName(), information.getRequestUnit())
+                        .onErrorResume(throwable ->
+                            exceptionHandler("Failed to create container", throwable))
                         .doOnNext(cosmosContainerResponse ->
                             fillAndProcessResponseDiagnostics(responseDiagnosticsProcessor,
                                 cosmosContainerResponse, null));
                 })
                 .block();
-        if (response == null) {
-            throw new CosmosDBAccessException("Failed to create collection");
-        }
+        assert response != null;
```

Comment:
The response could be null here.

Reply:
The response cannot be null here. 
assert is used to make sure, if ever the response is null, then it is a SDK bug and we should fix it. 
In an ideal case, it should never be null.